Add development environment setup guide for Fedora 28.

RyanFleck opened the pull request.

karfai commented on setup/fedora.md:
```diff
@@ -0,0 +1,193 @@
+
+Configuring these repositories will prepare all prerequisites for contributing to the project by improving the rule interpreter.
+
+## What is an Xalgorithm Rule?
```
Remove this section and change it to refer to definitions that we have elsewhere in documentation. We don't exactly have this description in a single document. If there's something missing that you've documented here, open an issue to extend the particular document.

RyanFleck replied:
Sure. This was originally written to be a blog post, I'll cut some of the exposition out.

karfai commented on setup/fedora.md:
```diff
@@ -0,0 +1,193 @@
+## Installation
+
+After cloning these repositories and installing dependencies, all files that are required to build the compiler, run rules, and contribute to the interpreter will be installed.
+
```
These should be references to README in the top-level repo (https://github.com/Xalgorithms/interlibr). Otherwise, we have to keep these component descriptions up to date with those in that file.

RyanFleck commented:
1. Steps provided for Ubuntu 18.04, did a test run on a VM this evening.
2. Fixed a small error in the Fedora docs, addressed the problems indicated by @karfai.
